.github/pull_request_template.md

```diff
@@ -57,5 +57,11 @@ If there are user-facing changes then we may require documentation to be updated
 -->
 
 <!--
-If there are any breaking changes to public APIs, please add the `breaking-change` label.
--->
\ No newline at end of file
+If there are any breaking changes to public APIs, please uncomment the following line:
+-->
+<!-- **This PR includes breaking changes to public APIs.** -->
+
+<!--
+If the changes fix either a security vulnerability or a bug that caused incorrect or invalid data to be produced. This is intended to mark changes that may affect users without their knowledge. For this reason, fixing bugs that cause errors or crashes don't count, since those are usually obvious.
+-->
```

Comment:
This is a little confusing to me.

If we were previously giving the wrong answer, and we fixed something to give a new answer, would I mark it as both breaking change and critical fix?

Also, hopefully most security fixes would go unnoticed by a user. I wouldn't expect them (by default) to change the output or the API.

Also, do we want to carve out an exception for experimental APIs?

Reply:
Hmm I guess I should clarify since Rok had the same question. The difference is "Breaking Change" is when we change our promise about how an API behaves, while "Critical Fix" changes the implementation to align with the existing promise. For example, if we have a function `add(x, y)` that promised to add two numbers, but we found that `add(2, 2)` returned `5`, fixing that to return `4` is a Critical Fix, not a breaking change. In the Arrow project, this typically shows up in file readers and writers, where we sometimes parse or serialize data incorrectly. Fixing those bugs don't really constitute breaking changes IMO.

Another way to put it is "Breaking Changes" are the reasons you might not want to upgrade until you have time to adapt your code, while "Critical Changes" highlight the risk of *not* upgrading.

Reply:
> Also, do we want to carve out an exception for experimental APIs?

And that's a good question on experimental APIs... I'm inclined to say it's still good to mark them as breaking changes. When we mark them as experimental, we are telling users to expect the API to change, but not that we won't warn them if they do 😄. That being said, in the future if we decide to do semantic versioning, we would probably treat breaking changes in experimental APIs differently from those in stable APIs.

Reply:
If we wanted, we could consider a separate tag for `Experimental API Change`.

Reply:
> Another way to put it is "Breaking Changes" are the reasons you might not want to upgrade until you have time to adapt your code, while "Critical Changes" highlight the risk of not upgrading.

This description is very helpful, let's include some of it.

> If we wanted, we could consider a separate tag for Experimental API Change.

No, I think your rationale is ok.  Even in our experimental APIs we usually avoid breaking changes if we can so it isn't like they occur all that often.